pkg/api/handlers/libpod/manifests.go

```diff
@@ -48,17 +50,18 @@ func ManifestCreate(w http.ResponseWriter, r *http.Request) {
 func ManifestInspect(w http.ResponseWriter, r *http.Request) {
 	runtime := r.Context().Value("runtime").(*libpod.Runtime)
 	name := utils.GetName(r)
-	newImage, err := runtime.ImageRuntime().NewFromLocal(name)
-	if err != nil {
-		utils.ImageNotFound(w, name, err)
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+	inspectReport, inspectError := imageEngine.ManifestInspect(r.Context(), name)
+	if inspectError != nil {
+		utils.Error(w, "Something went wrong.", http.StatusNotFound, inspectError)
 		return
 	}
-	data, err := newImage.InspectManifest()
-	if err != nil {
-		utils.InternalServerError(w, err)
+	var list manifest.Schema2List
+	if err := json.Unmarshal(inspectReport, &list); err != nil {
```

Comment:
Does this need to handle OCI index format?  If so, does it need to convert the index to a Docker manifest list, or should it just pass through what it got back unmodified?

If the manifest's type is not already in the Docker manifest list format, and we need to return a result in that format, we'd want to use `manifest.ListFromBlob()` to produce a list object, call the list's `ConvertToMIMEType()` method to get a second list object of the `manifest.DockerV2ListMediaType` type, and then call the second list's `Serialize()` method to re-encode it as JSON.

Reply:
Add the manifest type check in pkg/domain/infra/abi/manifest.go.